io/pci/PowerNVEEH.py

```diff
@@ -65,13 +65,21 @@ def setUp(self):
         cmd = "echo %d > /sys/kernel/debug/powerpc/eeh_max_freezes"\
             % self.max_freeze
         process.system(cmd, ignore_status=True, shell=True)
-        self.function = str(self.params.get('function'))
-        self.err = str(self.params.get('err'))
+        self.function = str(self.params.get('function'), default='4')
         self.pci_device = str(self.params.get('pci_device', default=' '))
         self.phb = self.pci_device.split(":", 1)[0]
         self.addr = genio.read_file("/sys/bus/pci/devices/%s/"
                                     "eeh_pe_config_addr" % self.pci_device)
         self.addr = str(self.addr).rstrip()
+        self.err = 0
+        for line in process.system_output('lspci -vs %s' % self.pci_device,
+                                          ignore_status=True,
+                                          shell=True).splitlines():
+            if 'Memory' in line and '64-bit, prefetchable' in line:
+                self.err = 1
```

Comment:
there are certain instances where, its 64bit and non-prefect...

 # lspci -v | grep -i "64-bit"
    Memory at 6000000000000 (64-bit, prefetchable) [size=32M]
    Memory at 6000002000000 (64-bit, prefetchable) [size=32M]
    Memory at 600c080100000 (64-bit, non-prefetchable) [size=64K]
    Memory at 6004000000000 (64-bit, prefetchable) [size=8M]
    Memory at 6004002000000 (64-bit, prefetchable) [size=32K]
    Memory at 6004000800000 (64-bit, prefetchable) [size=8M]
    Memory at 6004002010000 (64-bit, prefetchable) [size=32K]
    Memory at 6004001000000 (64-bit, prefetchable) [size=8M]
    Memory at 6004002020000 (64-bit, prefetchable) [size=32K]
    Memory at 6004001800000 (64-bit, prefetchable) [size=8M]
    Memory at 6004002030000 (64-bit, prefetchable) [size=32K]
    Memory at 600c100100000 (64-bit, non-prefetchable) [size=64K]
    Memory at 600c180000000 (64-bit, non-prefetchable) [size=1M]
    Memory at 600c180180000 (64-bit, non-prefetchable) [size=1K]
    Memory at 600c280000000 (64-bit, non-prefetchable) [size=64K]
    Memory at 600c280010000 (64-bit, non-prefetchable) [size=8K]
    Memory at 6200000000000 (64-bit, prefetchable) [size=32M]
    Memory at 6200002000000 (64-bit, prefetchable) [size=32M]
    Memory at 620c080100000 (64-bit, non-prefetchable) [size=64K]
    Memory at 620c100100000 (64-bit, non-prefetchable) [size=64K]
    Memory at 620c180000000 (64-bit, non-prefetchable) [size=16K]
    Memory at 6220000000000 (64-bit, prefetchable) [size=1M]

Reply:
```

Reply:
In 32 bit also we have both non-prefetchable and prefetchable, does both comes under mode 0. Are you referring any document, can u please share.

Reply:
@vrbagalkote will share in private.